src/OpenApi/src/Extensions/ApiDescriptionExtensions.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.OpenApi.Models;
+
+internal static class ApiDescriptionExtensions
+{
+    /// <summary>
+    /// Maps the HTTP method of the ApiDescription to the OpenAPI <see cref="OperationType"/> .
+    /// </summary>
+    /// <param name="apiDescription">The ApiDescription to resolve an operation type from.</param>
+    /// <returns>The <see cref="OperationType"/> associated with the given <paramref name="apiDescription"/>.</returns>
+    public static OperationType TopOperationType(this ApiDescription apiDescription) =>
+        apiDescription.HttpMethod switch
+        {
+            "GET" => OperationType.Get,
```

Comment:
These seem like things AspNetCore would expose constants for?

Reply:
We do expose constants in [here](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/HttpMethods.cs) but they are not compile-time constants so they can't be used in this switch expression. 😢